modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro

```diff
@@ -1,14 +1,126 @@
 definition {
 
+	var headers = '''
+			-u test@liferay.com:test \
+			-H Content-Type: application/json \
+		''';
+
+	macro _curlNObjectEntriesBody {
+		Variables.assertDefined(parameterList = "${fieldName},${fieldValue}");
+
+		var body = '''"${fieldName}": "${fieldValue}"''';
+
+		if (isSet(secondField)) {
+			var body = StringUtil.add("${body},", "\"secondField\":\"${secondFieldValue}\"", "");
+		}
+
+		if (isSet(externalReferenceCode)) {
+			var body = StringUtil.add("${body},", "\"externalReferenceCode\": \"${externalReferenceCode}\"", "");
+		}
+
+		if (isSet(nestedField)) {
```

Comment:
this could be moved to a separate macro, as there are a lot of variables inside that we don't assertDefined

Reply:
good idea, I will divide this into two methods in macro.